core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.directive.html

```diff
@@ -49,7 +49,7 @@
           <li class="nav-item"> <a ng-keydown="$ctrl.onMenuKeypress($event, 'aboutMenu', {tab: $ctrl.ACTION_CLOSE})" class="dropdown-item nav-link oppia-navbar-tab-content" href="https://oppiafoundation.org/volunteer" target="_blank" rel="noopener" translate="I18N_TOPNAV_GET_INVOLVED"></a></li>
         </ul>
       </li>
-      <li ng-if="SHOW_CLASSROOM_CALLOUT" ng-class="{'open' : $ctrl.activeMenuName === 'classroomMenu'}" uib-dropdown ng-show="$ctrl.navElementsVisibilityStatus.I18N_TOPNAV_CLASSROOM" aria-hidden="false" class="nav-item  oppia-navbar-clickable-dropdown protractor-test-classroom-oppia-list-item">
+      <li ng-class="{'open' : $ctrl.activeMenuName === 'classroomMenu'}" uib-dropdown ng-show="$ctrl.navElementsVisibilityStatus.I18N_TOPNAV_CLASSROOM && ($ctrl.CLASSROOM_PAGE_IS_SHOWN || $ctrl.updateShowClassroom())" aria-hidden="false" class="nav-item  oppia-navbar-clickable-dropdown protractor-test-classroom-oppia-list-item">
```

Comment:
How `ng-if="SHOW_CLASSROOM_CALLOUT"` was working without `$ctrl`?

Reply:
Before these changes, the constant was bound to `$scope` instead of `ctrl`.